src/components/intern-app.html

```diff
@@ -1,137 +1,179 @@
+           type: Array,
+           value: () => localStorageApi.getUsers()
+         },
+         addUserSelected: {
+           type: Boolean,
+           value: false
+         },
+         isExpanded: {
+            type: Boolean,
+            value: true
           },
-          addUserSelected: {
+         cardsPopulated: {
             type: Boolean,
             value: false
-          }
-        }
-      }
+         }
+       }
+     }
 
-      constructor() {
-        super();
-        this.boundNewUser = this.newUser.bind(this);
-      }
+     constructor() {
+       super();
+       this.boundNewUser = this.newUser.bind(this);
+     }
 
-      connectedCallback() {
-        super.connectedCallback();
-        window.addEventListener("newUser", this.boundNewUser);   
-      }
+     connectedCallback() {
+       super.connectedCallback();
+       window.addEventListener("newUser", this.boundNewUser);
+     }
 
-      disconnectedCallback() {
-        super.disconnectedCallback();
-        window.removeEventListener("newUser", this.boundNewUser); 
-      }
+     disconnectedCallback() {
+       super.disconnectedCallback();
+       window.removeEventListener("newUser", this.boundNewUser);
+     }
 
-      newUser(event) {
-        var newUserProfile = Object.assign({}, event.detail);
-        newUserProfile.id = this.genId();
-        this.pushToArray(this.userProfiles, newUserProfile);
-        this.addUserSelected = false;
-      }
+     deleteAll() {
+      if (window.confirm("Are you sure you want to delete all users?")) {
+        localStorageApi.deleteUsers();
+        cardsPopulated = false;
+        }  
+      } 
 
-      pushToArray(userProfiles, newUserProfile) {
-        var existingIds = userProfiles.map((newUserProfile) => newUserProfile.id);
-        if (!existingIds.includes(newUserProfile.id)) {
-          this.push('userProfiles', newUserProfile);
-        }
-        else {
-          newUserProfile.id = this.genId();
-          this.push('userProfiles', newUserProfile);
-        };
-      };
+     collapseAll() {}
 
-      genId() {
-        return '' + Math.random().toString(9).substr(2, 3);
-      }
+     newUser(event) {
+       const newUserProfile = Object.assign({},event.detail);
+       newUserProfile.id = localStorageApi.genId();
+       this.pushToArray(this.userProfiles, newUserProfile);
+       this.addUserSelected = false;
+       this.cardsPopulated = true;
+     }
 
-      createNewUserProfile(event) {
-        this.addUserSelected = !this.addUserSelected;
-      }
+     pushToArray(userProfiles, newUserProfile) {
+       const existingIds = userProfiles.map(newUserProfile => newUserProfile.id); 
+       const validUser = localStorageApi.validateUser(newUserProfile);
+       if (validUser) {
+         localStorageApi.createUser(newUserProfile);
+         this.push('userProfiles', newUserProfile);
+       }
+       else {
+         this.pushToArray(this.userProfiles, newUserProfile);
```

Comment:
isn't this causing an infinite loop?

Reply:
it shouldn't because it's generating a new id inside pushToArray()

Reply:
but you're not doing anything to change `newUserProfile`. So if it comes in once, and comes in invalid (according to `validUser`), it's just going to continue looping through this `pushToArray` function.

Reply:
nevermind, I get what's going on now. the resetting of the ID in `validateUser` is an odd choice. not to reset it, but to do so in a function called "validateUser". I would just it expect to to evaluate whether or not the object is valid, not make it valid if it's not.

Reply:
I'll see if that works better in another place. I agree it's not the best choice.

Reply:
I removed the `.genId()` that was being called from validateUser in localStorage so it's only returning true or false and only generating an id initially when the user is created.